include/vast/Conversion/Common/Rewriter.hpp

```diff
@@ -62,6 +67,24 @@ namespace vast::conv
             bld.setInsertionPointToEnd( block );
             return bld.template create< Trg >( std::forward< Args >( args ) ... );
         }
+
+        template< typename T >
+        void safe_erase( gap::generator< T > &&range )
+        {
+            // It is highly probable there is an ongoing iteration
+            // inside the generator, therefore we first freeze the value.
+            // TODO(c++23): Refactor.
+            std::vector< T > to_erase;
+            std::ranges::move(range.begin(), range.end(), std::back_inserter(to_erase));
```

Comment:
you know the purpose of ranges is not to write `begin` and `end` :P 

`std::ranges::move(range, std::back_inserter(to_erase));` should do

Also I suggest to rename `range` to `ops`.

Reply:
Tell that to few old days younger version of yourself that forgot to even include the `ranges::` subpart of the namespace :P Will fix it, did not realise that I am stepping over to ranges proper instead of old-style api.